SleepEngine/Engine/Components/RigidBody.h

```diff
@@ -0,0 +1,30 @@
+// Copyright 2019 Tihran Katolikian
+
+#pragma once
+
+BEGIN_NAMESPACE_SLEEP
+
+class RigidBody : public Component
+{
+public:
+    using Base = Component;
+
+    enum class Type: unsigned char
+    {
+          Rectangle = 0
+        , Ellipse
+    };
+
+    RigidBody() noexcept(true);
+
+    GET_SET(glm::vec2, getSize, setSize, m_size)
+    GET_SET(glm::vec2, getVelocity, setVelocity, m_velocity)
```

Comment:
wait, the type of member can be deduced in macro via decltype(). i thought you had that implemented. Necessary args are name of function and member only

Reply:
No, it doesn't work with class members.

Reply:
You could have mentioned it in the previous merge request btw :)